mapbox_vector_tile/__init__.py

```diff
@@ -14,12 +14,22 @@ def encode(
     extents=4096,
     on_invalid_geometry=None,
     check_winding_order=True,
+    crs_from=None,
+    crs_to=None,
 ):
-    vector_tile = encoder.VectorTile(extents, on_invalid_geometry, check_winding_order=check_winding_order)
+    vector_tile = encoder.VectorTile(
+        extents=extents,
+        on_invalid_geometry=on_invalid_geometry,
+        check_winding_order=check_winding_order,
+        quantize_bounds=quantize_bounds,
+        y_coord_down=y_coord_down,
+        crs_from=crs_from,
+        crs_to=crs_to,
+    )
```

Comment:
I like having these specified here, because imho the overwhelmingly common case is to share the same configuration across all layers in the tile. However, hoisting these options up means that we give up allowing per layer differences. Would it be reasonable to keep these at the top level here, but also allow per layer overrides?

Reply:
> For encoding several tiles with different behaviour, a simple concatenation is sufficient:

did you mean "layer" here instead of tile? Just to make sure I'm following you, we could achieve per layer differences in a single tile by encoding each separately as their own tile first and concatenating the payloads together to merge into a single tile, due to how protobufs work?

> The problem is that when decoding the "sum" of these two elements (with potentially several extents, y_coord_down, etc), I will need to give a dictionary (indexed by layer_name) to apply different options per layer.

Good call out, I  neglected to consider the decode case, which doesn't offer any option per layer. I was assuming for encode that callers would effectively perform the same operations as in the top level [encode](https://github.com/tilezen/mapbox-vector-tile/blob/2dfeb11db9cc40165a14b32c6285ddf0357b3dcb/mapbox_vector_tile/__init__.py#L18-L24) function but with different options as necessary when adding each layer.

Whatever option we choose, it's important for it to be symmetric between encode/decode. I can think of 3 possible options, which may not all be viable:

1. allow options to specify differences per layer to support encoding/decoding through a single call. Encode/decode options should be the same unless there's a good reason or them to diverge.
2. For per layer differences, callers add the layers one at a time, and extract the layers one at a time as well by name. We'd likely need to also grow a call to expose the list of layer names beforehand to support decoding.
3. If tiles can be generated by concatenating separate layer payloads, can we rely on the same mechanism to extract them out this way as well? Similar to 2, I assume we'd also need to provide a way to surface the ordering of layer names.

My preference would be option 1, since my expectation is that this would keep the call site code simplest. However, I think we'd need to think through supporting various scenarios, eg I think it'd be important for the per layer differences to be thought of as overrides, otherwise it'd be unclear how to decode layers that weren't specified. We could default to whatever we do now, but it might make more sense for callers to specify that fallback when there are per layer differences.

Maybe that implies that we leave the options the way they are now, and there's a separate catch all  to configure any option more specifically for each layer? I expect per layer options to be the minority case, so keeping the caller complexity isolated to that case seems justified. eg`decode(..., y_coord_down=True, per_layer=dict(layer_foo=dict(y_coord_down=False))`

Reply:
> did you mean "layer" here instead of tile?

Yes, I mean "layer". Sorry for the confusion.

I also prefer the option 1. I will try to do something like the option you propose.
`decode(..., y_coord_down=True, per_layer=dict(layer_foo=dict(y_coord_down=False))`